src/vs/workbench/contrib/terminal/browser/terminalEditorService.ts

```diff
@@ -153,31 +165,71 @@ export class TerminalEditorService extends Disposable implements ITerminalEditor
+	getOrCreateEditorInput(instanceOrUri: ITerminalInstance | SerializedTerminalEditorInput | URI, isFutureSplit: boolean = false): TerminalEditorInput {
+		let resource: URI;
+		if (URI.isUri(instanceOrUri)) {
+			resource = instanceOrUri;
+		} else if ('instanceId' in instanceOrUri) {
+			resource = instanceOrUri.resource;
+		} else {
+			resource = URI.parse(instanceOrUri.resource);
 		}
+		const inputKey = resource.path;
+		const cachedEditor = this._editorInputs.get(inputKey);
 		if (cachedEditor) {
 			return cachedEditor;
 		}
 
-		if ('pid' in instance) {
-			instance = this._terminalInstanceService.createInstance({ attachPersistentProcess: instance }, TerminalLocation.Editor);
+		if ('pid' in instanceOrUri) {
+			instanceOrUri = this._terminalInstanceService.createInstance({ attachPersistentProcess: instanceOrUri }, TerminalLocation.Editor);
+		}
+
+		// Terminal from a different window
+		if (URI.isUri(instanceOrUri)) {
+			const terminalIdentifier = parseTerminalUri(instanceOrUri);
+			if (terminalIdentifier.instanceId) {
+				this._primaryOffProcessTerminalService.requestDetachInstance(terminalIdentifier.workspaceId, terminalIdentifier.instanceId).then(attachPersistentProcess => {
+					const instance = this._terminalInstanceService.createInstance({ attachPersistentProcess }, TerminalLocation.Editor, resource);
+					input.setTerminalInstance(instance);
+					// TODO: fix and use an event instead, for now is used to trigger TerminalEditor's setInput,
+					// which attaches to the element
+					this._editorService.openEditor(input, {
+						pinned: true,
+						forceReload: true
+					},
+						input.group
+					);
+					this._registerInstance(inputKey, input, instance);
+				});
+			}
 		}
 
-		const input = this._instantiationService.createInstance(TerminalEditorInput, instance);
-		instance.target = TerminalLocation.Editor;
-		this._editorInputs.set(instance.instanceId, input);
-		this._instanceDisposables.set(instance.instanceId, [
-			instance.onDisposed(this._onDidDisposeInstance.fire, this._onDidDisposeInstance),
-			instance.onDidFocus(this._onDidFocusInstance.fire, this._onDidFocusInstance)
+		let input: TerminalEditorInput;
+		if ('instanceId' in instanceOrUri) {
+			instanceOrUri.target = TerminalLocation.Editor;
+			input = this._instantiationService.createInstance(TerminalEditorInput, resource, instanceOrUri);
+			this._registerInstance(inputKey, input, instanceOrUri);
+		} else {
+			input = this._instantiationService.createInstance(TerminalEditorInput, instanceOrUri, undefined);
+			this._registerInstance(inputKey, input);
+		}
+		return input;
+	}
+
+	private _registerInstance(inputKey: string, input: TerminalEditorInput, instance?: ITerminalInstance): void {
+		this._editorInputs.set(inputKey, input);
+		if (!instance) {
+			return;
+		}
+		this._instanceDisposables.set(inputKey, [
+			instance.onDidFocus(this._onDidFocusInstance.fire, this._onDidFocusInstance),
+			// TODO: why don't these do anything
```

Comment:
resolved?

Reply:
not resolved - we now call this._terminalEditorService.detachInstance(instanceToDetach); within terminal service on line 265

Reply:
Ok in that case let's remove this